packages/datetime/src/dateRangeInput.tsx

```diff
@@ -304,10 +342,16 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
 
     private handleStartInputChange = (e: React.FormEvent<HTMLInputElement>) => {
         this.handleInputChange(e, DateRangeBoundary.START);
+        if (this.props.startInputProps != null) {
+            Utils.safeInvoke(this.props.startInputProps.onChange, e);
+        }
     }
 
     private handleEndInputChange = (e: React.FormEvent<HTMLInputElement>) => {
         this.handleInputChange(e, DateRangeBoundary.END);
+        if (this.props.endInputProps != null) {
```

Comment:
oh wait you're already doing this! then why bother with the `!= null` check at all? i suspect that if you pass explicit `null` or `undefined` (falsy values), React would turn that into `{}` from defaultProps.